src/components/profile/ShareableForm.js

```diff
@@ -23,40 +23,44 @@ export default class ShareableForm extends PureComponent {
     event.preventDefault();
     const { shareableType, onComplete } = this.props;
     let { name, expiration, priority } = this.state;
+    // WAT?
     if(priority) {
       priority = 2;
     }
     if(expiration) {
       expiration = localizeDate(expiration);
     }
     const submission = { name, expiration, priority, type: shareableType };
-    onComplete(submission);
-    this.setState({
-      name: '',
-      expiration: '',
-      priority: false
-    });
+    onComplete(submission)
+      // wait for save to complete before clearing the form...
+      .then(() => {
+        this.setState({
+          name: '',
+          expiration: '',
+          priority: false
+        });
+      });
   };
 
   render() {
     const { action, shareableType } = this.props;
     const { name, expiration, priority } = this.state;
-
+    // Generally in components, don't use id's
```

Comment:
Does that mean we shouldn't use "htmlFor" to attach a label to an input? (Aren't they connected by id?)

Reply:
Can be okay if you know only one instance of form. :) Issue is with components generally trying not to make assumption about how many times component appears. One alternative is to wrap the `label` around the `input`. Same functionality as `htmlFor` without having to give an `id`